### DISCUSSION TITLE
wire:model doesn't set values of fields with type="hidden"

### DISCUSSION THREAD

**Discussion by sphkress** [livewire/livewire#6599] [7. Bugs (archived)] [answered]
### Livewire version

3.0.1

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

I want to use a Livewire component to set a value in a regular form. So i am simulating an extended dropdown selector. On click on a select list item, that attribute is set within the component. To pass this attribute value to the form, I use a hidden input field, linked with wire:model, but the value is not passed to the hidden field. It's only set, if its type is not 'hidden", e.g. text. It doesn't make sense to use CSS to hide it separately, e.g. absolute positioning or others.

$address_id is properly defined including 
```
    protected $rules = [
        'address_id'    => ['required', 'integer'],
    ];

    public function setValue($value) {
         $this->address_id=$value;
    }
```

Value for "address"  is **not** set
`<input type="hidden" wire:model="address_id" name="address">`

Value for "address" **is** set and will be passed on form submission
`<input type="text" wire:model="address_id" name="address">`
but requires further layout treatment

If this was a feature, it's well hidden, as searching for "site:livewire.laravel.com hidden" doesn't show any related information to this issue.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by koltejas** [comment]
Facing the same issue, really don't understand why. 
I had to use Alpine x-model with entangle to get around this. As livewire works so closely with alpine, it should support hidden inputs directly the same way.
Really hope this is supported.

**Comment by selfsimilar** [comment]
This is a regression. I just encountered this in a form I recently migrated from Livewire v2 to v3. Livewire v2 would happily set a hidden input value, but v3 won't. It's not semantically sound HTML, but like you mentioned, you can just use CSS to hide the input like `<input wire:model="address_id" name="address" style="display:none" />`

**Comment by sphkress** [comment, marked as answer]
Finally I found a solution which is logically understandable in my view:

I was using the hidden field only as a 'receiver' of data - a hidden field is usually not meant to be a source, rather being prefilled in a way and reused later. So the totally valid way of implementing it would be as this:

`<input type="hidden" value="{{ $address_id }}" name="address">`

**Comment by selfsimilar** [comment]
@sphkress while I agree that many `type="hidden"` inputs are as you described - static over the life of the form - not all are. I encountered this in several places that don't conform to that description. For example I have a 3rd-party javascript library for file upload that will return some values I need captured in the form post-upload. Previously I parsed the output in PHP and it flowed to e.g. `<input type="hidden" wire:model.live='mediaId'>`, but I had to change that to `<input style="display:none" wire:model.live='mediaId'>` to get those values on submission. While the [HTML Spec](https://html.spec.whatwg.org/multipage/input.html#hidden-state-(type=hidden)) says "The [input](https://html.spec.whatwg.org/multipage/input.html#the-input-element) element [represents](https://html.spec.whatwg.org/multipage/dom.html#represents) a value that is not intended to be examined or manipulated by the user.", I would say that the user is NOT manipulating the value, the code is.

At the very least, there is an argument to be made that `hidden` type inputs could hold dynamic data that is not directly user-manipulated, and that `wire:model` shouldn't just ignore those inputs. And also, this is behavior that, if intentional, should have been recorded in the migration documentation as it's a departure from the v2 behavior.
